File: 9/python/main.py
def solve(limit: int) -> int:
    max_a_b = int(limit / 2)
    for a in range(1, max_a_b):
        for b in range(1, max_a_b):
            c = limit - a - b
            if a**2 + b**2 == c**2 and a < b < c:
                print('Found', a, b, c)
                return a * b * c
    return None


if __name__ == '__main__':
    print('Answer =', solve(1000))

File: 9/go/main.go
package main

func checks(a int, b int, c int) bool {
	if a*a+b*b == c*c {
		if a < b && b < c {
			return true
		}
	}
	return false
}

func solve(limit int) int {
	var max_a_b int = int(limit / 2)
	for a := 1; a < max_a_b; a++ {
		for b := 1; b < max_a_b; b++ {
			var c int = limit - a - b
			if checks(a, b, c) {
				return a * b * c
			}
		}
	}
	return 0
}

func main() {
	println(solve(1000))
}
